src/plugins/kernel_install/utils.sh

```diff
@@ -85,20 +96,27 @@ function list_installed_kernels()
   output=$(echo "$output" | grep recovery -v | grep with | awk -F" " '{print $NF}')
 
   while read -r kernel; do
-    if [[ -f "$prefix/boot/vmlinuz-$kernel" && ! "$kernel" =~ .*\.old$ ]]; then
+    if [[ -f "$prefix/boot/vmlinuz-$kernel" ]]; then
+      if [[ -z "$all" ]]; then
+        [[ "$kernel" =~ .*\.old$ ]] && continue
+        cmd="sudo grep -q '$kernel' '$INSTALLED_KERNELS_PATH'"
+        cmd_manager 'SILENT' "$cmd" || continue
+      fi
       available_kernels+=("$kernel")
     fi
   done <<< "$output"
 
-  echo
+  if [[ "${#available_kernels[@]}" -eq 0 ]]; then
+    echo 'None of the installed kernels are managed by kw.' \
+      'Pass --list-all|-a to see all installed kernels'
+    return 0
```

Comment:
Maybe I missed something, but did you add a test that passes through this code path?

Reply:
The execution of the code was tested, but not the parsing of the option! Added in new version.